src/contract.rs

```diff
@@ -55,20 +57,31 @@ pub mod execute {
         deps: DepsMut,
         env: Env,
         deposit_ids: Vec<u32>,
+        profit_taking_or_stop_loss: Vec<bool>,
```

Comment:
As written here (and most other places) I cannot tell what this `bool` is meant to represent (what does `true` mean? That we're both profit taking and stop loss? That we're one or the other. Reading just this line I think I would assume `true` means `stop_loss` but later it turns out that is not the case). Anyway the normal way to do this in rust is via an enum. So something like

```
enum Reason {
    ProfitTaking,
    StopLoss,
}
```

Then this becomes a `Vec<Reason>` and it just overall makes more sense. But this is part of the public API (I think?) so maybe you just want a `profit_taking: bool`.

Reply:
I'd like to leave it as bool type to match with Vyper contract type.

Reply:
Fair. I still would prefer it were renamed.